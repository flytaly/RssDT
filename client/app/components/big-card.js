import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import BigCardHeader from './big-card-header';
import AddFeedModal from './add-feed-modal';

const StyledBigCard = styled.div`
    display: flex;
    position: relative;
    flex-direction: column;
    background: ${props => props.theme.cardBackground};
    color: ${props => props.theme.fontColor};
    width: ${props => props.theme.bigCardWidth}rem;
    max-width: calc(100vw - 2rem);
    min-height: 53rem;
    margin: 1rem auto;
    border-radius: 10px;
`;

const BigCard = ({ page, children }) => (
    <StyledBigCard id="bigCard">
        <BigCardHeader page={page} />
        {children}
        <AddFeedModal />
    </StyledBigCard>
);

BigCard.propTypes = {
    page: PropTypes.string.isRequired,
    children: PropTypes.element.isRequired,
};

export default BigCard;
