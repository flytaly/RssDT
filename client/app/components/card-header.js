import styled from 'styled-components';
import Link from 'next/link';
import PropTypes from 'prop-types';
import withAuth from './decorators/withAuth';

const Menu = styled.ul`
    display: flex;
    font-size: 1.3rem;
    list-style: none;
    margin: 0;
    padding: 0;
    align-self: flex-end;
    li {
        margin-left: 1rem;
    }
`;

const CardHeader = ({ me }) => (
    <Menu data-testid="card-header">
        <li><a href="/">Add new feed</a></li>
        {!me
            ? <li><Link href="/login"><a href="/login">Log in</a></Link></li>
            : (
                <>
                    <li><Link href="/feeds/manage"><a href="/feeds/manage">Manage</a></Link></li>
                    <li><Link href="/logout"><a href="/logout">Log out</a></Link></li>
                </>)}
    </Menu>
);

CardHeader.propTypes = {
    me: PropTypes.shape({}),
};

CardHeader.defaultProps = {
    me: null,
};

export default withAuth(false)(CardHeader);
