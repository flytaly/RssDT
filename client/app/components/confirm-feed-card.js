import React, { useState, useEffect } from 'react';
import { useMutation } from 'react-apollo-hooks';
import gql from 'graphql-tag';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import StyledCard from './styled/card';
import CardHeader from './card-header';
import GraphQLError from './graphql-error';
import Container from './styled/card-inner-container';

const CONFIRM_SUBSCRIPTION_MUTATION = gql`
  mutation CONFIRM_SUBSCRIPTION_MUTATION(
    $token: String!
  ) {
    confirmSubscription(
        token: $token
    ) {
        message
    }
  }
`;

const Message = styled.div`
    margin-top: 4rem;
    color: ${props => props.color};
`;

const ConfirmFeed = ({ token }) => {
    const [message, setMessage] = useState('Confirming subscription...');
    const [errorMsg, setErrorMsg] = useState('');
    const [confirmSubscription] = useMutation(CONFIRM_SUBSCRIPTION_MUTATION);

    useEffect(() => {
        async function confirm() {
            try {
                const { data } = await confirmSubscription({ variables: { token } });
                setMessage(data.confirmSubscription.message);
            } catch (e) {
                setErrorMsg(e.message);
                setMessage('');
            }
        }

        confirm();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [token]);

    return (
        <StyledCard>
            <Container>
                <CardHeader />
                <Message color={errorMsg ? 'red' : 'inherit'} data-testid="confirm-message">
                    {errorMsg ? <GraphQLError error={errorMsg} /> : message}
                </Message>
            </Container>
        </StyledCard>
    );
};

ConfirmFeed.propTypes = {
    token: PropTypes.string,
};

ConfirmFeed.defaultProps = {
    token: '',
};

export default ConfirmFeed;
export { CONFIRM_SUBSCRIPTION_MUTATION };
