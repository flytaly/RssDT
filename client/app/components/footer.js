import React from 'react';
import styled from 'styled-components';
import Link from 'next/link';

const StyledFooter = styled.footer`
    display: flex;
    flex-direction: column;
    margin-bottom: 0.5rem;
`;

const FooterRow = styled.div`
    display: inline-block;
    font-size: 1.3rem;
    color: ${({ theme }) => theme.footerFontColor};
    text-align: center;
    margin-top: 0.5rem;
    > span {
        margin-right: 0.5rem;
        vertical-align: middle;
    }
    svg {
        vertical-align: middle;
    }
    a:active,
    a:visited,
    a:link,
    a {
        text-decoration: none;
        color: inherit;
    }
    a:hover {
        color: ${({ theme }) => theme.footerHoverLinkColor};
    }
`;

const Footer = () => {
    const currentYear = (new Date()).getFullYear();
    return (
        <StyledFooter>
            <FooterRow>
                <span><Link href="/help"><a>Help</a></Link></span>
                <span><a href="mailto:support@feedmailu.com">Contact</a></span>
            </FooterRow>
            <FooterRow>
                <span>{currentYear}</span>
                <span>
                    <a href="https://github.com/flytaly/">
                        <span>Vitaly Yerofeyevsky</span>
                    </a>
                </span>
            </FooterRow>
        </StyledFooter>
    );
};

export default Footer;
