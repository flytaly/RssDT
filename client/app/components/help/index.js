import React from 'react';
import styled from 'styled-components';
import Sticky from '../sticky';
import contentList from './content-list';

const HelpPageContainer = styled.div`
    position: relative;
    width: 100%;
    font-size: 1.4rem;
    line-height: 1.4;
    h2 {
        font-size: 1.7em;
        text-align: center;
    }
`;

const TableOfContent = styled.nav`
    top: 10px;
    padding: 0.8rem;
    @media all and (min-width: 550px) {
        position: ${({ isFixed }) => (isFixed ? 'fixed' : 'absolute')};
        width: 16rem;
    }
    ul {
        margin: 0;
        padding-left: 1rem;
    }
    a {
        text-decoration: none;
        color: grey;
    }
    a:hover {
        color: black;
    }
`;

const Content = styled.div`
    @media all and (min-width: 550px) {
        margin-left: 16rem;
    }
    h3 {
        font-size: 1.4em;
        margin-top: 4rem;
    }
    img {
        display: block;
        margin: 2rem auto;
        max-width: 100%;
        box-shadow: 0rem 0px 10px 0px rgba(0,0,0,0.5);
        border-radius: 4px;
    }
`;

const Help = () => (
    <HelpPageContainer>
        <h2>Help</h2>
        <Sticky>
            {isFixed => (
                <TableOfContent isFixed={isFixed}>
                    <ul>
                        {contentList.map(({ id, headerText }) => (
                            <li key={id}>
                                <a href={`#${id}`}>{headerText}</a>
                            </li>))}
                    </ul>
                </TableOfContent>)}
        </Sticky>
        <Content>
            {contentList.map(({ id, headerText, content }) => (
                <article key={id}>
                    <h3 id={id}>{headerText}</h3>
                    {content}
                </article>))}
        </Content>
    </HelpPageContainer>);

export default Help;
