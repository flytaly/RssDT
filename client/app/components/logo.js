import React from 'react';
import styled from 'styled-components';
import Link from 'next/link';
import logoBlackIcon from '../static/icon-black.png';

const LinkWithLogo = styled.a`
        display: inline-flex;
        align-items: center;
        border-radius: 10px;
        background-color: rgba(255,255,255, 0.66);
        padding: 0.2rem 1rem;
        color: inherit;
        text-decoration: none;
`;

const Icon = styled.img`
    width: 2.5rem;
    height: 2.5rem;
    margin-right: 1rem;
`;

const Title = styled.span`
    font-weight: bold;
    font-size: 1.8rem;
    margin-right: 1rem;
`;
const Description = styled.span`
    color: dimgray;
    font-size: 1.2rem;
`;

const Logo = () => (
    <Link href="/">
        <LinkWithLogo href="/">
            <Icon src={logoBlackIcon} />
            <div>
                <Title>FeedMailu</Title>
                <Description>feed to email aggregator</Description>
            </div>
        </LinkWithLogo>
    </Link>
);

export default Logo;
