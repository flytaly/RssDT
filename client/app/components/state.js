/**
 * This is simple implementation of 'redux-like' state manager
 * based on the article: https://medium.com/front-end-weekly/react-hooks-tutorial-for-pure-usereducer-usecontext-for-global-state-like-redux-and-comparison-dd3da5053624
 */
import React, { createContext, useContext, useReducer } from 'react';
import PropTypes from 'prop-types';

export const initialState = {
    newFeedModal: { isOpen: false },
    isBrowser: false,
};
export const types = {
    toggleNewFeedModal: 'toggleNewFeedModal',
    setIsBrowser: 'setIsBrowser',
};

export const reducer = (state, action) => {
    switch (action.type) {
        case types.toggleNewFeedModal: return {
            ...state,
            newFeedModal: { isOpen: !state.newFeedModal.isOpen },
        };
        case types.setIsBrowser: return {
            ...state,
            isBrowser: true,
        };
        default: return state;
    }
};

export const stateCtx = createContext(initialState);
const dispatchCtx = createContext(() => null);

export const useDispatch = () => useContext(dispatchCtx);

export const useGlobalState = (property = null) => {
    const state = useContext(stateCtx);
    return property ? state[property] : state;
};

export const StateProvider = ({ children }) => {
    const [state, dispatch] = useReducer(reducer, initialState);
    return (
        <dispatchCtx.Provider value={dispatch}>
            <stateCtx.Provider value={state}>
                {children}
            </stateCtx.Provider>
        </dispatchCtx.Provider>
    );
};

StateProvider.propTypes = {
    children: PropTypes.node.isRequired,
};
