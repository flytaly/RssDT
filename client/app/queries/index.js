// ---- QUERIES ----
export { default as ME_QUERY } from './me-query';
export { default as MY_FEEDS_QUERY } from './my-feeds-query';
export { default as MY_FEED_ITEMS_QUERY } from './my-feed-items-query';
export { default as MY_FEED_ITEMS_CONNECTION } from './my-feed-items-connection-query';

// ---- MUTATIONS ----
export { default as UPDATE_MY_INFO_MUTATION } from './update-my-info-mutation';
