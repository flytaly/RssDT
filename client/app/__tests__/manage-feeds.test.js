import '@testing-library/jest-dom/extend-expect';
import { render, wait, queryHelpers, act } from '@testing-library/react';
// import { act } from 'react-dom/test-utils';
import ManageFeeds from '../components/manage-feeds';
import { MY_FEEDS_QUERY } from '../queries';
import ApolloMockedProvider from '../test-utils/apollo-mocked-provider';
import types from '../types/digest-periods';
import { StateProvider } from '../components/state';
import Page from '../components/page';

const getByNameAttribute = queryHelpers.queryByAttribute.bind(
    null,
    'data-name',
);

const createdAt = new Date('2019-04-19T00:11:00.000Z');
const lastUpdate = new Date('2019-04-19T00:12:00.000Z');
const mocks = [{
    request: { query: MY_FEEDS_QUERY },
    result: {
        data: {
            myFeeds: [{
                __typename: 'UserFeed',
                id: 'id1',
                createdAt,
                lastUpdate,
                schedule: types.DAILY,
                activated: true,
                withContentTable: 'DEFAULT',
                feed: {
                    id: 'testFeedId',
                    title: 'Test Feed Title',
                    link: 'http://testfeed.com',
                    url: 'http://testfeed.com/feed',
                    imageTitle: 'Image Title',
                    imageUrl: 'http://testfeed.com/logo.png',
                    __typename: 'Feed',
                },
            }],
        },
    },
}];

describe('Manage feeds', () => {
    const component = (
        <Page>
            <ApolloMockedProvider mocks={mocks}>
                <StateProvider>
                    <ManageFeeds />
                </StateProvider>
            </ApolloMockedProvider>
        </Page>);

    const getContainer = async () => {
        let container;
        await act(async () => {
            ({ container } = render(component));
        });
        return container;
    };

    test('should match snapshot', async () => {
        const container = await getContainer();
        await wait(() => {
            expect(container.firstChild).toMatchSnapshot();
        });
    });

    test('should render date and time in current locale', async () => {
        const container = await getContainer();
        const createdAtLocale = new Date(createdAt).toLocaleDateString();
        const lastUpdateLocale = new Date(lastUpdate).toLocaleString();
        await wait(() => {
            expect(getByNameAttribute(container, 'ADDED')).toHaveTextContent(createdAtLocale);
            expect(getByNameAttribute(container, 'LAST DIGEST DATE')).toHaveTextContent(lastUpdateLocale);
        });
    });
});
