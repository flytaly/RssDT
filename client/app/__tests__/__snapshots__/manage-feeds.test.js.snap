// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage feeds should match snapshot 1`] = `
.c8 {
  margin: 0px;
  padding: 0px;
  font-size: 1.3rem;
}

.c9 {
  list-style: none;
  display: grid;
  grid-template-columns: 6fr 2fr 4fr 4fr 1fr;
  grid-gap: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.4rem 0;
}

.c9:hover:not(:first-of-type) {
  outline: rgba(115,40,109,0.1) 1px solid;
  background-color: rgba(115,40,109,0.1);
}

.c12 {
  min-width: 10rem;
  word-break: break-all;
}

.c13 {
  min-width: 8rem;
  word-break: break-all;
}

.c10 {
  font-size: 1.5rem;
  font-weight: bold;
  min-width: 10rem;
}

.c11 {
  font-size: 1.5rem;
  font-weight: bold;
  min-width: 8rem;
}

.c14 {
  border: none;
  padding: 0;
  background: none;
  text-align: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin: 0 0.5rem 0 0;
}

.c14:focus,
.c14:hover {
  -webkit-transform: scale(1.10);
  -ms-transform: scale(1.10);
  transform: scale(1.10);
}

.c14:focus:hover {
  outline: none;
}

.c14:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c14 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.c6 {
  padding: 0 2rem 1rem;
}

.c7 {
  font-size: 1.8rem;
}

.c15 {
  background-color: hsl(0,0%,60%);
  color: white;
  padding: 0.6rem 1rem;
  border: none;
  text-align: center;
  text-transform: uppercase;
  border-radius: 5px;
  min-width: 7rem;
  font-size: 1.3rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: hsl(200,100%,40%);
  margin-top: 1rem;
}

.c15:focus,
.c15:hover {
  background-color: hsl(0,0%,45%);
}

.c15:active {
  color: hsl(0,0%,90%);
}

.c15:focus,
.c15:hover {
  background-color: hsl(200,100%,30%);
}

.c15:active {
  color: hsl(200,100%,60%);
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  background-color: rgba(255,255,255,0.66);
  padding: 0.2rem 1rem;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 1rem;
}

.c4 {
  font-weight: bold;
  font-size: 1.8rem;
  margin-right: 1rem;
}

.c5 {
  color: dimgray;
  font-size: 1.2rem;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 0.5rem;
}

.c18 {
  display: inline-block;
  font-size: 1.3rem;
  color: #b2b2b2;
  text-align: center;
  margin-top: 0.5rem;
}

.c18 > span {
  margin-right: 0.5rem;
  vertical-align: middle;
}

.c18 svg {
  vertical-align: middle;
}

.c18 a:active,
.c18 a:visited,
.c18 a:link,
.c18 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c18 a:hover {
  color: #dddddd;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  padding: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c16 {
  -webkit-flex: 1 0 1rem;
  -ms-flex: 1 0 1rem;
  flex: 1 0 1rem;
}

@media all and (max-width:37em) {
  .c9 {
    grid-template-columns: 1fr;
    padding: 1rem 0;
  }

  .c9:nth-child(2n+1):not(:first-of-type):not(:hover) {
    background-color: rgba(115,40,109,0.03);
  }
}

@media all and (max-width:37em) {
  .c12 {
    display: grid;
    grid-template-columns: 40% 1fr;
    grid-gap: 5px;
  }

  .c12:before {
    content: attr(data-name);
    font-size: 1.5rem;
    font-weight: bold;
  }
}

@media all and (max-width:37em) {
  .c13 {
    display: grid;
    grid-template-columns: 40% 1fr;
    grid-gap: 5px;
  }

  .c13:before {
    content: attr(data-name);
    font-size: 1.5rem;
    font-weight: bold;
  }
}

@media all and (max-width:37em) {
  .c10 {
    display: none;
  }
}

@media all and (max-width:37em) {
  .c11 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/"
    >
      <img
        alt="Logo"
        class="c3"
        src="/static/icon-black.png"
      />
      <div>
        <span
          class="c4"
        >
          FeedMailu
        </span>
        <span
          class="c5"
        >
          feed to email aggregator
        </span>
      </div>
    </a>
    <main
      class="c6"
    >
      <h1
        class="c7"
      >
        Edit subscriptions
      </h1>
      <ul
        class="c8"
      >
        <li
          class="c9"
        >
          <div
            class="c10"
          >
            FEED
          </div>
          <div
            class="c11"
          >
            ADDED
          </div>
          <div
            class="c10"
          >
            LAST DIGEST DATE
          </div>
          <div
            class="c10"
          >
            DIGEST SCHEDULE
          </div>
          <div
            class="c11"
          >
            ACTIONS
          </div>
        </li>
        <li
          class="c9"
        >
          <div
            class="c12"
            data-name="FEED"
          >
            <a
              href="/feeds/view?id=id1"
              title="View items"
            >
              Test Feed Title
            </a>
          </div>
          <div
            class="c13"
            data-name="ADDED"
          >
            <span />
          </div>
          <div
            class="c12"
            data-name="LAST DIGEST DATE"
          >
            <span />
          </div>
          <div
            class="c12"
            data-name="DIGEST SCHEDULE"
          >
            Daily digest
          </div>
          <div
            class="c13"
            data-name="ACTIONS"
          >
            <div>
              <button
                class="c14"
                title="Edit the feed"
                type="button"
              >
                <svg
                  data-file-name="SvgEdit"
                />
              </button>
              <button
                class="c14"
                title="Delete the feed"
                type="button"
              >
                <svg
                  data-file-name="SvgTrash"
                />
              </button>
            </div>
          </div>
        </li>
      </ul>
      <button
        class="c15"
        type="button"
      >
        Add new feed
      </button>
    </main>
    <div
      class="c16"
    />
    <footer
      class="c17"
    >
      <div
        class="c18"
      >
        <span>
          <a
            href="/help"
          >
            Help
          </a>
        </span>
        <span>
          <a
            href="mailto:support@feedmailu.com"
          >
            Contact
          </a>
        </span>
      </div>
      <div
        class="c18"
      >
        <span>
          2019
        </span>
        <span>
          <a
            href="https://github.com/flytaly/"
          >
            <span>
              Vitaly Yerofeyevsky
            </span>
          </a>
        </span>
      </div>
    </footer>
  </div>
</div>
`;
